shap/explainers/other/__init__.py

```diff
@@ -1,5 +1,13 @@
-from ._coefficent import Coefficient
+import warnings
+
+from ._coefficient import Coefficient
 from ._lime import LimeTabular
 from ._maple import Maple, TreeMaple
 from ._random import Random
 from ._treegain import TreeGain
+
+
+# Deprecated class alias with incorrect spelling
+def Coefficent(*args, **kwargs):  # noqa
+    warnings.warn("Coefficent has been renamed to Coefficient", DeprecationWarning)
+    return Coefficient(*args, **kwargs)
```

Comment:
Let's also set a version for when we will remove this deprecated class entirely, and mention it in the warning. How about `0.45.0`?

Reply:
Good idea, done.